### ISSUE TITLE
PHP Version Issue with Invoice Ninja 4.5.46

### ISSUE THREAD

**Issue by andrewhaji** [invoiceninja/dockerfiles#416]
I run Invoice Ninja 4 through Docker Compose.

I recently upgraded to 4.5.46, and now the app container won't start. The log files show a PHP version mismatch - it says it requires PHP 7.3.0 or newer, but it's only running 7.2.0. Is there some way to fix this?

Let me know if log files would be helpful and I can post them.

Thanks,
Andrew

**Comment by turbo124** [comment]
@andrewhaji 

i've kicked off another build 4.5.46-p1 with PHP 7.4

**Comment by andrewhaji** [comment]
Thanks for the quick reply!

I'm now experiencing a different issue. The docker containers are running but when I access the site, I get a 500 error.

Server logs are showing:

```
production.ERROR: ***ErrorException*** [0] : /var/www/app/app/Http/Middleware/StartupCheck.php [Line 73] => Use of undefined constant NINJA_VERSION - assumed 'NINJA_VERSION' (this will throw an Error in a future version of PHP)
```

Many thanks for your help,
Andrew

**Comment by turbo124** [comment]
@andrewhaji It looks like the container breaks when attempting to upgrade from PHP 7.2 to either 7.3 or 7.4, I'll need to chat with @lwj5 to see whether we can get this fixed.

**Comment by andrewhaji** [comment]
Is there any way to work around this issue? Can I take a backup of my data, tear down my Docker setup, and then start from scratch? Any advice would be most appreciated.

**Comment by turbo124** [comment]
@andrewhaji this is a separate issue we've noticed on some different configurations. 

the fix here is to uncomment line 37 in public/index.php

**Comment by andrewhaji** [comment]
Thank you for your reply.

I'm trying to test that fix, but line 37 of `public/index.php` isn't anything meaningful in IN 4.5.46. I do see the commented-out line in IN 4.5.48. Could you push that version up to Docker so that I can test this?

Thank you!

**Comment by turbo124** [comment]
The issue with .48 is that it requires PHP 7.4 and we haven't been able to upgrade the container yet unfortunately

**Comment by arwaldman** [comment]
I'm experiencing the same issue. I haven't been able to get the migration from V4 to V5 to work on 4.5.45, so I attempted to update my docker container to the latest 4.5.46-p2 and get the same 500 error with the same message in the logs.

@turbo124 Do you have any idea when the .48 container might be made available on Docker?

**Comment by andrewhaji** [comment]
FWIW, I did the following:

1. Took a backup of my Invoice Ninja database
2. Set up a fresh instance of IN 4.5.48 using Softaculous on my web hosting account
3. Imported the database

...and I am _still_ getting a 500 error. So I think this issue goes deeper than just the Docker containers.

I wasn't able to find any useful logs on my web server, but if there's anything I can help test, let me know.

**Comment by turbo124** [comment]
@andrewhaji 

There should be errors in the logs in storage/logs.

you may need to turn on APP_DEBUG=true in the .env file

**Comment by andrewhaji** [comment]
Thanks for the recommendation. I enabled APP_DEBUG and discovered that the issue I was having was related to my APP_KEY having been regenerated. I copied the APP_KEY over from my Docker install and it seems to be working.

I can keep my install running this way until the issue with PHP in Docker is figured out. Though, I may just migrate to IN5 in the next little while, anyway.

Thanks for all your work on this fine project.
